Add/connection verified errors react initial state

chrisbliss18 opened the pull request:
#### Changes proposed in this Pull Request:
Add the connection verified errors to the connection React initial state. This is part of a bigger project (1202697549597858-as) to provide consistent failed connection error reporting and an interface to reconnect (1202697549597858-as-1202697549597864). 

The changes will affect the following:
* Jetpack
* Connection

#### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion
* p9dueE-5w8-p2

#### Does this pull request change what data or activity we track or use?
* No

#### Testing instructions:
* Go to Jetpack > Dashboard.
* Open up the browser's developer console.
* Input `JP_CONNECTION_INITIAL_STATE` and press enter.
* Verify that the JS object that is output has a member named `connectionVerifiedErrors`, and that it's empty.
* Go to "Jetpack Debug -> XML-RPC Errors" and create a verified error.
* Go back to the Jetpack Dashboard and confirm that `JP_CONNECTION_INITIAL_STATE.connectionVerifiedErrors` now contains the newly created error.

chrisbliss18 committed:
Add: Verified errors to React initial state

chrisbliss18 committed:
changelog

chrisbliss18 committed:
Fix: Changed to calling method statically

Fixed incorrectly calling a static method non-statically. Added unit
testing for the Initial_State class.

chrisbliss18 committed:
Ran tools/fixup-project-versions.sh

chrisbliss18 committed:
[not verified] Merge branch 'trunk' into add/connection-verified-errors-react-initial-state

chrisbliss18 committed:
Merge branch 'trunk' into add/connection-verified-errors-react-initial-state

chrisbliss18 committed:
Updated hashes in plugin lock files.

chrisbliss18 committed:
Merge branch 'trunk' into add/connection-verified-errors-react-initial-state

sergeymitr committed:
Merge branch 'trunk' into add/connection-verified-errors-react-initial-state

chrisbliss18 committed:
Merge branch 'trunk' into add/connection-verified-errors-react-initial-state

chrisbliss18 committed:
Fixed versions

chrisbliss18 committed:
Merge branch 'trunk' into add/connection-verified-errors-react-initial-state

chrisbliss18 committed:
Merge branch 'trunk' into add/connection-verified-errors-react-initial-state

chrisbliss18 committed:
Merge branch 'trunk' into add/connection-verified-errors-react-initial-state

chrisbliss18 committed:
Merge branch 'trunk' into add/connection-verified-errors-react-initial-state

sergeymitr commented:
Hi @chrisbliss18.
The code looks good, and everything works perfectly.
I'm only worried that `connectionVerificationErrors` might imply that the errors come from some verification process, while the those are "verified" (i.e. confirmed) connection errors. May be "verified" would be a better fit, what do you think?

chrisbliss18 committed:
Update: Adjusted wording to be more consistent

Changed `connectionVerificationErrors` to `connectionVerifiedErrors` as
that's in keeping with the wording from `get_verified_errors()`, which
is where it is pulling the data from.

chrisbliss18 commented:
> Hi @chrisbliss18. The code looks good, and everything works perfectly. I'm only worried that `connectionVerificationErrors` might imply that the errors come from some verification process, while the those are "verified" (i.e. confirmed) connection errors. May be "verified" would be a better fit, what do you think?

Agreed. I just pushed up a change to reflect that.

sergeymitr reviewed (approved):
👍
